### ISSUE TITLE
Missing DocumentOrShadowRoot interface implementation

### ISSUE THREAD

**Issue by bedeoverend** [skatejs/cloudydom#5]
_See https://github.com/webcomponents/shadydom/issues/113._

It seems only `activeElement` is implemented - it'd be great to have support for all methods / props in the interface. My use case is using Prosemirror which relies on the interface for managing selections. I'd be keen to have a crack at a PR, but don't have time right now.

cc/ @treshugart - looks like you guys had this exact [problem](https://bugs.webkit.org/show_bug.cgi?id=163921) with WebKit, did you ever get around it?

**Comment by matthewp** [comment]
I'm not too familiar with these APIs as I don't use them often. Happy to accept a PR of course!

**Comment by bedeoverend** [comment]
Fair enough - I'm not too familiar either, but I'll take a crack when I can. Interestingly, the first Shadow DOM polyfill [looks](https://github.com/webcomponents/webcomponentsjs/blob/v0/src/ShadowDOM/wrappers/ShadowRoot.js#L73) to have just called `document.getSelection()` for `shadowRoot.getSelection()`, might be able to be applied for a few of the methods in this interface 😕

**Comment by treshugart** [comment]
> Shadow DOM polyfill looks to have just called document.getSelection() for shadowRoot.getSelection(), might be able to be applied for a few of the methods in this interface 😕

I think that's probably fine. I know things like `ShadowRoot.activeElement` is filled by simply doing `document.activeElement` because there's no encapsulation.

**Comment by bedeoverend** [comment]
Yeah though I'm not sure if there'd be repercussions of returning the document selection if that selection isn't encapsulated by the shadow root

**Comment by treshugart** [comment]
Yeah, you'd probably have to test something like `shadowRoot.contains(document.getSelection().anchorNode)`.
